posthog/models/insight.py

```diff
@@ -120,3 +120,15 @@ def dashboard_item_saved(sender, instance: Insight, dashboard=None, **kwargs):
         filter = get_filter(data=instance.dashboard_filters(dashboard=dashboard), team=instance.team)
 
         instance.filters_hash = generate_cache_key("{}_{}".format(filter.toJSON(), instance.team_id))
+
+
+class InsightViewed(models.Model):
```

Comment:
Great idea!

Reply:
and can also power "Popular Insights in your team", whenever we get to it!

Reply:
.. actually, the rfc seems to have explicitly mentioned not doing "views" as part of the activity log: https://github.com/PostHog/posthog/issues/8545 , because that gets into the territory of auditing what a user did.

Unsure if this has changed (cc: @pauldambra )

Reply:
Sidestepping that philosophising (although its an interesting question... and there does seem to be overlap)

Why not use ClickHouse? We're already tracking who views insights there... e.g. https://app.posthog.com/insights/tAVJZdGi (that doesn't seem to actually breakdown by distinct id but can be a fixed version of whatever query this API call is making)

Reply:
Otherwise if you put the calls to the InsightViewed model behind `log_activity` and teach that to write to `InsightViewed` for an activity of `viewed` and scope of `Insight`. And similar change in `load_activity`. Then we've separated the application's API for logging things that happen from the implementation of storage

E.g. I agree it's probably a good idea to clear the views periodically

But I'd rather read this from clickhouse than duplicate the storage...

Reply:
_or_ there's nothing stopping us from using the ActivityLog storage and clearing only `scope: Insight` and `activity:viewed` every N days

Then we keep a single API for logging activity

Reply:
@rcmarron stream of consciousness aside

I have to stay awake and active until 2200 my time to collect my daughter from a performance. Can sync up later if you like?

Reply:
A couple of thoughts here:

1. Right now, the data being stored here is targeted at enabling a "your most recently viewed" feature. With that in mind, it isn't tracking every view. Rather, it's tracking a 'last_viewed_at' timestamp, so if you view an insight five times, there's only one record. This should reduce the size of the table by an order of magnitude. The idea here is to track what we need for the application, but not recreate our analytics in postgres.
2. I know there are thoughts around building other features that could rely on this data, namely, "your team's most popular insights". Depending on the algorithm that we want to use for that feature, we may end up wanting to track each individual view. With that said, I don't want to preempt that decision/feature and unnecessarily store data that we don't need yet. By its nature, this data has a TTL, so it's not a big deal to change it down the road.